compiler/sem/semstmts.nim

```diff
@@ -1177,18 +1177,27 @@ proc semNormalizedConst(c: PContext, n: PNode): PNode =
       #      semantic analysis, also the fact that it likely implies bugs... :/
       case defPart.kind
       of nkVarTuple:
+        let def = producedDecl[^1]
+        # FIXME: the arity check happens too late, meaning that it's possible
+        #        for there to be an arity mismatch here, in which case the
+        #        `def` is accessed out-of-bounds (i.e. the compiler will crash)
         v.ast =
-          case def[i].kind
-          of nkExprColonExpr:
-            def[i][1]
-          else:
-            def[i]
+          case def.kind
+          of nkTupleConstr: skipColon(def[i])
+          of nkError:       def
+          else:             unreachable()
       of nkConstDef:
         # xxx: rework skConst to store the whole def like let/var
         v.ast = producedDecl[^1]
       else:
         internalError(c.config, "should never happen")
 
+    if v.ast.isError:
+      # XXX: although this mirrors the behaviour of ``semNormalizedLetOrVar``,
```

Comment:
Admittedly, the handling of type and symbols is really sloppy in the compiler in general and I sadly repeated this pattern.